proxy/ControlMatcher.cc

```diff
@@ -220,6 +220,7 @@ HostMatcher<Data, MatchResult>::NewEntry(matcher_line *line_info)
   if (error.failed()) {
     // There was a problem so undo the effects this function
     memset(cur_d, 0, sizeof(Data));
+    new(cur_d) Data(); //reconstruct
```

Comment:
This makes sense to me, but do we still need the memset()? It's obviously harmless if not necessary, mostly curious.